### DISCUSSION TITLE
Chargers from Charge Amps and their claimed OCPP 1.6J support

### DISCUSSION THREAD

**Discussion by tuomaz** [lbbrhzn/ocpp#816] [General]
Hi
Has anyone tried the claimed support of OCPP 1.6J of these chargers?

https://www.chargeamps.com/product/charge-amps-aura/
https://www.chargeamps.com/product/charge-amps-dawn/

**Comment by tuomaz** [comment]
I now have a Charge Amps Dawn installed. It connects and I see traffic between the charger and HA / OCCP. I will continue to test and see what works and not.

**Comment by tuomaz** [comment]
Example data, no charge ongoing (redacted som ID's?)
`homeassistant    | 2023-06-06 21:11:47.813 INFO (MainThread) [ocpp] XXX: receive message [2,"YYY","MeterValues",{"connectorId":1,"meterValue":[{"timestamp":"2023-06-06T19:11:47Z","sampledValue":[{"value":"75170","context":"Sample.Periodic","measurand":"Energy.Active.Import.Register","unit":"Wh"},{"value":"13.00","context":"Sample.Periodic","measurand":"Current.Offered","unit":"A"},{"value":"0.000","context":"Sample.Periodic","measurand":"Current.Import","phase":"L1","unit":"A"},{"value":"0.000","context":"Sample.Periodic","measurand":"Current.Import","phase":"L2","unit":"A"},{"value":"0.000","context":"Sample.Periodic","measurand":"Current.Import","phase":"L3","unit":"A"},{"value":"0.00","context":"Sample.Periodic","measurand":"Voltage","phase":"L1","unit":"V"},{"value":"0.00","context":"Sample.Periodic","measurand":"Voltage","phase":"L2","unit":"V"},{"value":"0.00","context":"Sample.Periodic","measurand":"Voltage","phase":"L3","unit":"V"}]}]}]`

**Reply by Bryneholt** [reply]
Have you been able to start the charging with this integration? 
im able to get some information, but unable to actually start the charging. 

2023-08-20 15:48:39.039 INFO (MainThread) [ocpp] xxxx: send [2,"xxx-446d-4a7b-a340-a81420eef280","RemoteStartTransaction",{"idTag":"xxxx","connectorId":1}]
2023-08-20 15:48:39.056 INFO (MainThread) [ocpp] xxxx: receive message [3,"xxx-446d-4a7b-a340-a81420eef280",{"status":"Accepted"}]

but nothing does happen,
From the car point of view, the charger says "paused" 


edit:

Because I had enabled RFID to charge, while using the Charge amps app, the charger still requires this.
Ie, I can only start it with the RFID tag.

**Reply by tuomaz** [reply]
I cannot stop the charging from Home Assistant. When I try to stop the charge with the "Charging control" switch, it immediately goes back to the on state. Does this work for you?

**Reply by Bryneholt** [reply]
Hello, I’ve used the workaround and created automations setting the current to 0 when to stop charging, and to 16A when to start the charging. 

This works. 

`alias: EV Smart Charging - Start
description: ""
trigger:
  - platform: state
    entity_id:
      - sensor.smartcharger_charging
    from: "off"
    to: "on"
condition: []
action:
  - service: number.set_value
    data:
      value: "16"
    target:
      entity_id: number.charger_maximum_current
mode: single
`

**Reply by Bryneholt** [reply]
And to stop

alias: EV Smart Charging - Stop
description: ""
trigger:
  - platform: state
    entity_id:
      - sensor.smartcharger_charging
    from: "on"
    to: "off"
condition: []
action:
  - service: number.set_value
    target:
      entity_id: number.charger_maximum_current
    data:
      value: "0"
mode: single

**Comment by Miggle007** [comment]
I have the Charge Amps Aura. I am able to connect and to charge to connectorId 1 only. I am not able to use the full functionality using this OCPP instance and also, as mentioned above, not all sensors are processing the right information. I have asked Charge Amps for an API plugging into their cloud service as a workaround.

**Comment by pelleL21** [comment]
I have a Charge Amps Halo. How did you guys get the OCPP to work? I have put the adress ws://homeassistant.local:9000 in the Halo charger, but it doesen't connect to the OCPP server.

**Reply by HenrikBurton** [reply]
I'm considering a Charge Amp Halo, can anyone confirm that it works with this plug-in?

**Reply by ClarreJu** [reply]
I have an old Halo.  
You have to give the ip-adress in numbers. ws://192.168.1.xx:9000

I got connected but only one measurand got a value. When locking in the debug log at HA I saw some relevant and correct readings but there where also some error message and all measurands where "Unavailable" except one. I gave up, didn't have time to sort it all out and switched back to the cloud. 
If any one has any ideas I love to test them!

**Reply by HenrikBurton** [reply]
Thank you for sharing. Could i have improved over time? On what firmware version are your Halo?

**Reply by ClarreJu** [reply]
Hmm... Worth checking. I'll switch back to local and let I be for a while. I'll get back. I have removed the integration so I can't see the firmware. I'll get back with that too.

**Reply by ClarreJu** [reply]
Firmware: 184

**Reply by HenrikBurton** [reply]
Ok, that's the most resent version from feb 2025 according to the [release note.](https://wwwchargeampscom.cdn.triggerfish.cloud/uploads/2025/02/Charge-Amps-Halo-fw184-release-notes.pdf) Perhaps I should reconsider buying the Halo. Thanks!

**Reply by ClarreJu** [reply]
The reason I get Unavailable is because the Halo crashes every time. I have to restart it by flipping of the fuses. Not so good. I give up and go back to the cloud.

**Reply by nymeron** [reply]
any follow ups on Halo integration to HA? 
Do I understand correctly, that you can either have the charger connected to their cloud or HA, but not both?

**Reply by HenrikBurton** [reply]
Correct, You can't have both.

I'm on version 189 and that works fine.

**Comment by Miggle007** [comment]
Just an extra thought:
I am using the ChargeAmps repo, also available via HACS: https://github.com/kirei/hass-chargeamps.

The advantage of using this is that the native cloud and app integeation still functions. Furthermore, I had some issues with my Chargeamps Aura and the connectorIDs (dual socket), which is not an issue with the Halo obviously.  

This OCPP integration was too high maintenance for me.

**Reply by htukiain** [reply]
Is your Chargeamps upgraded to OCPP? I can remotely start charging from HA, but ending and restarting does not work from HA. All works fine from the ChargeAmpps app. 

It seems that on/off and start / end charge might be different commands?

**Reply by nymeron** [reply]
can this integration be used to dynamically set current using `set_max_current` during the charging session?

**Reply by htukiain** [reply]
it can be used. Below is my HA-YAML for that (use only as reference / example)
<code>
alias: EV - Dynamic load balancer (stable)
description: >
  Uses Shelly EM3 phase currents and last commanded EV current to keep main fuse
  below limit. Requires input_number.ev_charger_setpoint.
triggers:
  - trigger: time_pattern
    seconds: /30
conditions:
  - condition: state
    entity_id: binary_sensor.ev_charging_real
    state: "on"
  - condition: state
    entity_id: timer.charger_cooldown
    state: idle
actions:
  - action: logbook.log
    data:
      name: EV-load-balancer
      message: >
        Tick │ ia {{ ia }} A, ib {{ ib }} A, ic {{ ic }} A │ max_phase {{
        max_phase }} A │ house_load {{ house_load }} A │ headroom_after_off {{
        headroom_after_off }} A │ ev_setpoint {{ ev_setpoint }} A → allowed_now
        {{ allowed_now }} A → target {{ target_amps }} A
  - choose:
      - conditions:
          - condition: template
            value_template: "{{ allowed_now|int == 0 and ev_setpoint|float(0) > 0 }}"
        sequence:
          - action: automation.trigger
            data:
              entity_id: automation.charge_amps_stop
          - action: timer.start
            target:
              entity_id: timer.charger_cooldown
          - action: logbook.log
            data:
              name: EV-load-balancer
              message: >
                Headroom after off {{ headroom_after_off }} A too low for even
                {{ min_allowed }} A + {{ safety_margin }} A margin → charger
                disabled, cooldown 15 min started.
      - conditions:
          - condition: template
            value_template: "{{ target_amps|int != current_limit|int and target_amps|int > 0 }}"
        sequence:
          - delay: "00:00:02"
          - action: chargeamps.set_max_current
            data:
              chargepoint: 2409085367M
              connector: 1
              max_current: "{{ target_amps|int }}"
          - action: logbook.log
            data:
              name: EV-load-balancer
              message: >
                Set charger to {{ target_amps }} A (phase load A/B/C: {{ ia
                }}/{{ ib }}/{{ ic }} A)
    default: []
  - action: input_number.set_value
    target:
      entity_id: input_number.ev_charger_setpoint
    data:
      value: "{{ target_amps|int }}"
  - action: logbook.log
    data:
      name: EV-load-balancer
      message: >
        Stored EV setpoint {{ target_amps }} A to
        input_number.ev_charger_setpoint
variables:
  fuse_limit: 25
  safety_margin: 1
  min_allowed: 6
  max_allowed: 16
  up_smoothing: 0.5
  ia: "{{ states('sensor.shellyem3_3494547b6480_channel_a_current')|float(0) }}"
  ib: "{{ states('sensor.shellyem3_3494547b6480_channel_b_current')|float(0) }}"
  ic: "{{ states('sensor.shellyem3_3494547b6480_channel_c_current')|float(0) }}"
  ev_setpoint: "{{ states('input_number.ev_charger_setpoint')|float(0) }}"
  max_phase: "{{ [ia, ib, ic]|max }}"
  house_load: "{{ [0, max_phase - ev_setpoint]|max }}"
  headroom_after_off: "{{ fuse_limit|float(0) - house_load|float(0) }}"
  raw_allowed_total: "{{ headroom_after_off|float(0) - safety_margin|float(0) }}"
  allowed_total_clamped: "{{ [max_allowed|float(0), raw_allowed_total]|min }}"
  allowed_now: |-
    {% if allowed_total_clamped < min_allowed %}
      0
    {% else %}
      {{ [max_allowed|float(0), allowed_total_clamped]|min | int }}
    {% endif %}
  current_limit: "{{ ev_setpoint|float(0) }}"
  target_amps: >-
    {% set c = current_limit %} {% set a = allowed_now|float(0) %} {% if a == 0
    %}
      0
    {% elif a > c %}
      {{ (c + (a - c) * up_smoothing) | round(0, 'ceil') | int }}
    {% elif a < c %}
      {{ a | int }}
    {% else %}
      {{ c | int }}
    {% endif %}
mode: queued
</code>

**Comment by kyrreaa** [comment]
I also have the ChargeAmps Halo, and they updated my firmware just a few days ago. (4.4-2025)
I can see a log message in Homeassistant saying "WARNING (MainThread) [py.warnings] /usr/local/lib/python3.13/site-packages/dataclasses_json/core.py:201: RuntimeWarning: 'NoneType' object value of non-optional type hardware_version detected when decoding ChargePoint" and I wonder if it is related to it not showing up?

**Reply by Niek** [reply]
I managed to connect my Aura using OCPP (ws://IP:9000), but nothing seems to work. All attributes are empty and nothing appears in the logs.

**Comment by robbionline** [comment]
I'll get the Charge Amps Luna installed tomorrow. Let's see, if I get it it running. Anyone here with some advice?

**Comment by HenrikBurton** [comment]
I installed a Charge Amps Halo (I love the form that fits my hours perfectly) despite the above issues and have got it working like I wanted. One issue is the occasionally it loses connection with the OCCP server in my HA but reconnects within a minute and the lose connection again after ~2 minutes. This can be fixed by restarting the Halo. Not ok but bearable until a fix is found.

**Comment by robbionline** [comment]
> I'll get the Charge Amps Luna installed tomorrow. Let's see, if I get it it running. Anyone here with some advice?

I had some issues, but mostly my own faults. The charge amps Luna works quite well now. I'll have to look why many sensors are not populated, but that may also be expected.

**Reply by perbjore** [reply]
How did you manage to connect with the luna? I've tried ws://192.168.xx.xx:9000 without any success, does it need to run against a secure OCPP instance?

**Comment by perbjore** [comment]
I had issues with my ChargeAmps luna not publishing Voltage and current correcly in the intregration for all phases. Sometimes it was set as attributes to Current.Import but it was very flaky.  I've fixed it in this branch here now all the metering values are working consistently as current_import_l1-l3 and voltage_l1-l3, have asked for merging to mainline here https://github.com/lbbrhzn/ocpp/pull/1839

If not you can use it directly from here in the meanwhile
https://github.com/perbjore/ocpp

**Comment by Xembalo** [comment]
I've successfully connected my Luna to Home Assistant; everything looks good so far—the sensors, etc., are working fine.

The Luna is running version 1.7.5, but version 2.0.1 is available online (https://www.chargeamps.com/firmware-updates/). Even after a week in the official cloud, it hasn’t updated. 


Has anyone managed to install the firmware update via OCPP yet? All attempts fail immediately with the message “Firmware upload status: DownloadFailed.” No matter where I place the file—I’ve tried various web servers on the LAN and the Internet, using both HTTP and HTTPS—the access log doesn’t even show that the Luna is attempting to connect to the host at all. The only thing I haven’t tried yet is Wireshark. 


Does anyone have any experience with this?


```
action: ocpp.update_firmware
data:
  devid: garage
  firmware_url: http://192.168.1.64:8080/luna_2.0.1.img.enc
  delay_hours: 1
```

**Reply by Xembalo** [reply]
It worked with ftp://192.168.1.64/luna_2.0.1.img.enc 👍
